x-pack/plugins/ml/public/application/components/anomalies_table/links_menu.tsx

```diff
@@ -104,6 +107,34 @@ export const LinksMenuUI = (props: LinksMenuProps) => {
     return location;
   };
 
+  const getAnomalySourceMapsLink = async (
+    anomaly: AnomaliesTableRecord,
+    sourceIndicesWithGeoFields: {
+      [key: string]: { [key: string]: { geoFields: string[]; dataViewId: string } };
+    }
+  ) => {
+    // Create a layer for each of the geoFields
+    const initialLayers = getInitialSourceIndexFieldLayer(
+      sourceIndicesWithGeoFields[anomaly.jobId]
+    );
+
+    const locator = share.url.locators.get(MAPS_APP_LOCATOR);
+    const location = await locator?.getLocation({
+      initialLayers,
+      timeRange: data.query.timefilter.timefilter.getTime(),
```

Comment:
We need to use the same time range behavior as the 'View in maps' action, where the start is the time of the anomalous bucket, and the end is `anomaly time + bucket span - 1ms`

Reply:
Happy to update that but we should keep in mind that sometimes, with smaller bucket spans, there may be no results on the map until the user expands the timerange.

Reply:
Updated in 194d15241a2663884443d87ef1f6505302ba654d